src/Sentry/SentrySdk.cs

```diff
@@ -28,13 +28,39 @@ public static class SentrySdk
         /// </summary>
         public static SentryId LastEventId { [DebuggerStepThrough] get => _hub.LastEventId; }
 
+        internal static IHub InitHub(SentryOptions options)
+        {
+            // Side-effects in a factory function 🤮
```

Comment:
Suggestion for an alternative approach? We want to setup logging before anything runs
```suggestion
```

Reply:
For example, instead of mutating `SentryOptions` here, we could have `SentryOptions.GetLoggerOrDefault()` that would return the user-configured logger or initialize a console logger. Essentially, change the behavior from push-based to pull-based.

Reply:
Although I concede that is rather nitpicky at this point because there are many other places where `SentryOptions` is mutated.

Reply:
The goal is to set an instance on the options so it's available throughout the SDK that's relying on `options.Logger?.LogX`